Update all existing eventing documentation to reflect removal of event-mesh (#10820)
kyma-project/kyma: docs/eventing/02-01-eventing-architecture.md

---
title: Architecture
---

Kyma Eventing uses NATS to implement an Event Publisher Proxy and Eventing Controller which work together to process and deliver events in Kyma.

## Event Publisher Proxy

> The Event Publisher Proxy component receives legacy and Cloud Event publishing requests from the cluster workloads (microservice or Serverless Functions) and redirects them to the NATS server. It also fetches a list of subscriptions for a connected application.

Reviewer: ```suggestion
The Event Publisher Proxy component receives legacy and Cloud Event publishing requests from the cluster workloads (microservices or Serverless Functions) and redirects them to the NATS server. It also fetches a list of subscriptions for a connected application.
```

> The Event Publisher Proxy component receives legacy and Cloud Event publishing requests from the cluster workloads (microservice or Serverless Functions) and redirects them to the Enterprise Messaging Service Cloud Event Gateway. It also fetches a list of subscriptions for a connected application.

Reviewer: The Event Publisher Proxy component receives legacy and Cloud Event publishing requests from the cluster workloads (microservice or Serverless Functions) and redirects them to the Enterprise Messaging Service Cloud Event Gateway. It also fetches a list of subscriptions for a connected application.
```suggestion
The Event Publisher Proxy component receives legacy and Cloud Event publishing requests from the cluster workloads (microservice or Serverless Functions) and redirects them to the NATS server. It also fetches a list of subscriptions for a connected application.
```

> The Event Publisher Proxy component receives legacy and Cloud Event publishing requests from the cluster workloads (microservice or Serverless Functions) and redirects them to the Enterprise Messaging Service Cloud Event Gateway. It also fetches a list of subscriptions for a connected application.

Reviewer: ```suggestion
The Event Publisher Proxy component receives legacy and Cloud Event publishing requests from the cluster workloads (microservice or Serverless Functions) and redirects them to the NATS server. It also fetches a list of subscriptions for a connected application.
```

## Eventing Controller

> The Eventing Controller component manages the internal infrastructure in order to receive an event. The Controller watches subscription Custom Resource Definitions. When an event is received in an Application, it lays down the Eventing infrastructure in NATS in order to trigger a Function.

Reviewer: Please mention that the NATS controller also dispatches Events to subscribers (microservices or Serverless Functions).

## Event types

> Kyma Eventing supports both cloud events and legacy events. Kyma converts legacy events to cloud events and adds the prefix `sap.kyma`.

Reviewer: ```suggestion
Kyma Eventing supports both cloud events and legacy events. Kyma converts legacy events to cloud events and adds the prefix `sap.kyma.custom`.
```

> For a Subscription Custom Resource, the fully qualified event type is in the form `sap.kyma.custom.commerce.order.created.v1`.

Reviewer: Could you please explain how the event type is composed?

```
The Event Type is composed of the following components:

Prefix: sap.kyma.custom
Application: commerce
Event: order.created
Version: v1
```

​
For publishers, it is:
- `order.created` for legacy-events coming from the `commerce` application

> - `sap.kyma.custom.commerce.order.cre`

Reviewer: ```suggestion
- `sap.kyma.custom.commerce.order.created.v1` for CloudEvents.
```

---
title: Architecture
---

> Kyma Eventing uses NATS to implement an Event Publisher Proxy and Eventing Controller which work together to process and deliver events in Kyma.

Author: In this sentence I said `an Event Publisher.... and Eventing Controller` so `an` was implied for the second one, but maybe `the` or no article at all would be better :) Thank you!
